pallets/stablewap/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+				id: pool_id,
+				who,
+				shares: amount,
+				amounts: (amounts.0, amounts.1),
+			});
+
+			Ok(())
+		}
+
+		/// Execute a swap of `asset_in` for `asset_out`.
+		///
+		/// Parameters:
+		/// - `origin`:
+		/// - `pool_id`: Id of a pool
+		/// - `asset_in`: ID of asset sold to the pool
+		/// - `asset_out`: ID of asset bought from the pool
+		/// - `amount`: Amount of asset sold
+		/// - `min_buy_amount`: Minimum amount required to receive
+		///
+		/// Emits `SellExecuted` event when successful.
+		///
+		#[pallet::weight(<T as Config>::WeightInfo::sell())]
+		#[transactional]
+		pub fn sell(
+			origin: OriginFor<T>,
+			pool_id: PoolId<T::AssetId>,
+			asset_in: T::AssetId,
+			asset_out: T::AssetId,
+			amount_in: Balance,
+			min_buy_amount: Balance,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				amount_in >= T::MinimumTradingLimit::get(),
+				Error::<T>::InsufficientTradingAmount
+			);
+
+			ensure!(
+				T::Currency::free_balance(asset_in, &who) >= amount_in,
+				Error::<T>::InsufficientBalance
+			);
+
+			let pool = Pools::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+
+			ensure!(pool.assets.contains(asset_in), Error::<T>::AssetNotInPool);
+			ensure!(pool.assets.contains(asset_out), Error::<T>::AssetNotInPool);
+
+			let pool_account = T::ShareAccountId::from_assets(&pool.assets, Some(POOL_IDENTIFIER));
+
+			let reserve_in = T::Currency::free_balance(asset_in, &pool_account);
+			let reserve_out = T::Currency::free_balance(asset_out, &pool_account);
+
+			let amount_out = calculate_out_given_in(
+				reserve_in,
+				reserve_out,
+				amount_in,
+				T::Precision::get(),
+				pool.amplification,
+			)
+			.ok_or(ArithmeticError::Overflow)?;
+
+			let fee_amount = Self::calculate_fee_amount(amount_out, pool.fee, true).ok_or(ArithmeticError::Overflow)?;
+
+			let amount_out = amount_out.checked_sub(fee_amount).ok_or(ArithmeticError::Underflow)?;
+
+			ensure!(amount_out >= min_buy_amount, Error::<T>::BuyLimitNotReached);
+
+			T::Currency::transfer(asset_in, &who, &pool_account, amount_in)?;
+			T::Currency::transfer(asset_out, &pool_account, &who, amount_out)?;
+
+			Self::deposit_event(Event::SellExecuted {
+				who,
+				asset_in,
+				asset_out,
+				amount_in,
+				amount_out,
+				fee: fee_amount,
+			});
+
+			Ok(())
+		}
+
+		/// Execute a swap of `asset_out` for `asset_in`.
+		///
+		/// Parameters:
+		/// - `origin`:
+		/// - `pool_id`: Id of a pool
+		/// - `asset_out`: ID of asset bought from the pool
+		/// - `asset_in`: ID of asset sold to the pool
+		/// - `amount`: Amount of asset sold
+		/// - `max_sell_amount`: Maximum amount allowedto be sold
+		///
+		/// Emits `buyExecuted` event when successful.
+		///
+		#[pallet::weight(<T as Config>::WeightInfo::buy())]
+		#[transactional]
+		pub fn buy(
+			origin: OriginFor<T>,
+			pool_id: PoolId<T::AssetId>,
+			asset_out: T::AssetId,
+			asset_in: T::AssetId,
+			amount_out: Balance,
+			max_sell_amount: Balance,
```

Comment:
well. this is consistent now with the xyk interface. where in case of buy - out is first. 

Same is done for omnipool. 

I would like to have in/out in same order everywhere too.

Reply:
cc @maht0rz WDYT as consumer of this?

Reply:
will be resolved in #469 if needed.